src/java.base/share/classes/jdk/internal/misc/CDS.java

```diff
@@ -195,4 +204,129 @@ private static void validateInputLines(String[] lines) {
         };
         return retArray;
     }
+
+    private static native void dumpClassList(String listFileName);
+    private static native void dumpDynamicArchive(String archiveFileName);
+
+    private static void outputStdStream(InputStream stream, String fileName, List<String> cmds) {
+        String line;
+        InputStreamReader isr = new InputStreamReader(stream);
+        BufferedReader rdr = new BufferedReader(isr);
+
+        try {
+            PrintStream prt = new PrintStream(fileName);
+            prt.println("Command:");
```

Comment:
[Try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) should be used to make sure the streams are closed when the method exits (normally or on exception).

```
try (InputStreamReader isr = new InputStreamReader(stream);
     BufferedReader rdr = new BufferedReader(isr);
     PrintStream prt = new PrintStream(fileName);)
{
    prt.println("Command:"); ....
```